client/state/lib/middleware.js

```diff
@@ -84,7 +82,7 @@ const receiveSelectedSitesChangeListener = ( dispatch, action ) => {
  */
 const removeSelectedSitesChangeListener = ( dispatch, action ) => {
 	debug( 'removeSelectedSitesChangeListener' );
-	delete selectedSiteListeners[ action.subscriberId ];
+	selectedSiteChangeListeners = selectedSiteChangeListeners.filter( listener => listener !== action.listener );
```

Comment:
Filter is fine here, but just a heads up that we don't polyfill all Array prototype methods